apps/emqx_utils/src/emqx_utils_calendar.erl

```diff
@@ -563,6 +564,14 @@ date_size(timezone) -> 5;
 date_size(timezone1) -> 6;
 date_size(timezone2) -> 9.
 
+dym(Y, M) ->
+    case is_leap_year(Y) of
+        true when M > 2 ->
+            dm(M) + 1;
+        _ ->
+            dm(M)
+    end.
```

Comment:
I prefer to use the implementation in the OTP's calendar module, as they also have some validations on date ranges.